python/pyspark/sql/catalog.py

```diff
@@ -29,31 +31,35 @@
 Column = namedtuple("Column", "name description dataType nullable isPartition isBucket")
 Function = namedtuple("Function", "name description className isTemporary")
```

Comment:
We should probably have these two (as well as `Database` and `Table`)  annotated here. If we're going to in-line all annotations, it will require choosing a convention. Personally, I prefer 3.6 syntax:

```python
class Coulumn(NamedTuple):
    name: str
    description: str
    dataType: str
    nullable: bool
    isPartition: bool
    isBucket: bool
```

Reply:
Sounds good! Actually I agree and prefer it.
If we would, we should do it in a separate PR to make it clear whether the change works as expected or not.

Reply:
Great! I create https://issues.apache.org/jira/browse/SPARK-36897 to track that. Thanks for the input!